sdk/react/src/views/AdPlaybackScreen/AdPlaybackScreen.js

```diff
@@ -244,13 +246,22 @@ export default class AdPlaybackScreen extends Component {
     return iconViews;
   };
 
-  render() {
+  componentDidUpdate(prevProps: Readonly<P>, prevState: Readonly<S>, snapshot: SS) {
     const isPastAutoHideTime = (new Date).getTime() - this.props.lastPressedTime > AUTOHIDE_DELAY;
     const doesAdRequireControls = this.props.ad && this.props.ad.requireControls;
     // TODO: IMA Ads UI is still not supported - No way to show UI while allowing Learn More in a clean way
     const isContent = !this.props.ad;
-    const shouldShowControls = this.props.screenReaderEnabled ? true : !isPastAutoHideTime && (doesAdRequireControls || isContent);
+    const isVisible = this.props.screenReaderEnabled ? true : !isPastAutoHideTime && (doesAdRequireControls || isContent);
 
+    if (isVisible !== this.state.shouldShowControls) {
```

Comment:
not clear why do you use state here

Reply:
i check previous state to learn if i have to pass information about ui